File: extension/dbus.js
// Заглушка для D-Bus общения
export class IziGhostDBus {
    constructor(extension) {
        this._extension = extension;
        console.log('IziGhost: D-Bus инициализирован (заглушка)');
    }

    sendChatMessage(text) {
        console.log('IziGhost D-Bus stub: отправка сообщения:', text);
    }

    triggerOcr(pngBytes) {
        console.log('IziGhost D-Bus stub: триггер OCR');
    }

    startListening() {
        console.log('IziGhost D-Bus stub: старт аудио');
    }

    stopListening() {
        console.log('IziGhost D-Bus stub: стоп аудио');
    }

    destroy() {
        console.log('IziGhost: D-Bus уничтожен');
    }
}

File: crates/common/src/dbus.rs
pub const DBUS_SERVICE_NAME: &str = "com.izighost.Daemon";
pub const DBUS_OBJECT_PATH: &str = "/com/izighost/Daemon";
pub const DBUS_INTERFACE_NAME: &str = "com.izighost.Daemon";

// В интерфейсе com.izighost.Daemon определены следующие методы:
// - StartRvms() -> u32 (pipewire_node_id)
// - StopRvms() -> ()
// - SendChatMessage(text: String) -> ()
// - TriggerOcr() -> ()
// - StartListening() -> ()
// - StopListening() -> ()
// - ListProfiles() -> Vec<String>
// - GetProfile(id: String) -> Profile
// - SaveProfile(profile: Profile) -> Profile
// - DeleteProfile(id: String) -> ()
// - SetActiveProfile(id: String) -> ()
// - GetActiveProfile() -> Option<Profile>
//
// И сигналы:
// - ChatChunk(delta_text: String)
// - ChatCompleted()
// - OcrCompleted(text: String)
// - AsrCompleted(text: String)
// - ErrorOccurred(message: String)
